tests/mock/dist.rs

```diff
@@ -466,6 +482,10 @@ fn create_tarball(relpath: &Path, src: &Path, dst: &Path) -> io::Result<()> {
             xzwriter = xz2::write::XzEncoder::new(outfile, 0);
             &mut xzwriter
         }
+        s if s.ends_with(".tar.zst") => {
```

Comment:
These blocks here look like an enum trying to break free

Reply:
Yeah, I wasn't terribly happy about duplicating this code, but again I think the refactor can come later.